static/js/chart/draw.ts

```diff
@@ -438,46 +469,85 @@ function drawLineChart(
 /**
  * Return all styles information for given dataGroupsDict.
  * {
- *     colors: [],
- *     dashes: [],
- *     statVars: [],
- *     geoIds: [],
- *     maxV: ,
+ *     colors: string[], an array of colors for statVars
+ *     dashes: string[], an array of dash styles for geoIds(if applied)
+ *     statVars: string[], an array of statVars
+ *     geoIds: string[], an array of geoIds
  * }
  *
  * @param dataGroupsDict
  */
-function generateStyles(
+function computePlotParams(
     dataGroupsDict: {[geoId: string]: DataGroup[] },
-) {
-  let styles = {};
-  styles["colors"] = [];
-  styles["dashes"] = [];
-  styles["statVars"] = [];
-  styles["geoIds"] = [];
+) : {
+  "colors": string[],
+  "dashes": string[],
+  "statVars": string[],
+  "geoIds": string[],
+} {
+  let plotParams : {
+    "colors": string[],
+    "dashes": string[],
+    "statVars": string[],
+    "geoIds": string[]};
+
+  plotParams = {
+    "colors": [],
+    "dashes": [],
+    "statVars": [],
+    "geoIds": []
+  };
 
   let dataGroups : DataGroup[];
   dataGroups = Object.values(dataGroupsDict)[0];
   let legendText = dataGroups.map((dataGroup) => dataGroup.label ? dataGroup.label: '');
   let colorFn = getColorFn(legendText);
 
   for (let i = 0; i < dataGroups.length; i++) {
-    styles["colors"].push(colorFn(dataGroups[i].label));
-    styles["statVars"].push(dataGroups[i].label);
+    plotParams["colors"].push(colorFn(dataGroups[i].label));
+    plotParams["statVars"].push(dataGroups[i].label);
   }
 
-  let dashIndex = 0;
+  for (let geoId in dataGroupsDict) {
+    plotParams["geoIds"].push(geoId);
+  }
+
+  plotParams["dashes"] = getDashes(plotParams["geoIds"]);
+
+  return plotParams;
+}
+/**
+ * Return a dict with max value and min value shown in the y label.
+ *  {
+ *      minV: number, max value in y label.
```

Comment:
It looks like you got the descriptions mixed up here :)

Reply:
Fixed~